src/CMakeLists.txt

```diff
@@ -105,7 +107,12 @@ if(ENABLE_RNTUPLE)
 endif()
 
 PODIO_ADD_LIB_AND_DICT(podioRootIO "${root_headers}" "${root_sources}" root_selection.xml)
-target_link_libraries(podioRootIO PUBLIC podio::podio ROOT::Core ROOT::RIO ROOT::Tree ROOT::ROOTVecOps)
+target_link_libraries(podioRootIO PUBLIC podio::podio
+                                         ROOT::Core
+                                         ROOT::RIO
+                                         ROOT::Tree
+                                         ROOT::ROOTVecOps
+                                         ROOT::ROOTDataFrame)
```

Comment:
I am contemplating whether we should move the `ROOTDataSource` into a separate library instead of making it part of the `podioRootIO` one. The main argument against it is that we will have one more library. However, if we really make use of the `Reader` interface internally, then we would need to link `podioRootIO` against `podioIO` (making a cyclic lib dependency) and potentially also against `podioSioIO`. Then we would have no separation of the libraries at all any longer. 

Since it is probably easier to merge libraries down the line, than splitting them apart, I would suggest to leave the `podioRootIO` library untouched and instead make a new `podioRDataSource` library that links against `podio::podioIO` (and all the necessary things from ROOT).

Reply:
Yes, I had to link against `podio::podioIO` in couple of places. Will create separate library...